app/src/main/java/com/example/myweatherapp/MainActivity.kt

```diff
@@ -18,19 +18,22 @@ import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.example.myweatherapp.databinding.ActivityMainBinding
+import com.example.myweatherapp.interfaces.IListItem
 import com.example.myweatherapp.interfaces.OnItemClickListener
 import com.example.myweatherapp.interfaces.OnSourceClickListener
 import com.example.myweatherapp.models.NewsModel
 import com.example.myweatherapp.repository.NewsRepository
+import java.util.ArrayList
 
 class MainActivity : AppCompatActivity(), OnItemClickListener, OnSourceClickListener {
 
-
     private var sharedPreferences: SharedPreferences? = null  // обращаемся к списку сразу после создания и получаем краш
 
     private lateinit var binding: ActivityMainBinding
     private val newsAdapter = NewsAdapter(this, this)
     private val repository = NewsRepository()
+    private val favoriteAdapter = FavoriteAdapter(this)
+    private lateinit var newsRepository: NewsRepository
```

Comment:
you have already defined the `NewsRepository` above. Why do you need two instances of the same object? ;)
also, lateinit is unnecessary here.

Reply:
removed repository-copy, it was added by chanse